src/uu/sleep/src/sleep.rs

```diff
@@ -61,15 +64,21 @@ pub fn uu_app() -> App<'static, 'static> {
         )
 }
 
-fn sleep(args: Vec<&str>) {
+fn sleep(args: Vec<&str>) -> UResult<()> {
     let sleep_dur =
-        args.iter().fold(
+        args.iter().try_fold(
             Duration::new(0, 0),
             |result, arg| match uucore::parse_time::from_str(&arg[..]) {
-                Ok(m) => m + result,
-                Err(f) => crash!(1, "{}", f),
+                Ok(m) => Ok(m + result),
+                Err(f) => Err(f),
```

Comment:
Great work on this! I'm just wondering whether we can merge these `match` expressions. Maybe something like this:
```rust
let sleep_dur =
    args.iter().try_fold(
        Duration::new(0, 0),
        |result, arg| match uucore::parse_time::from_str(&arg[..]) {
            Ok(m) => Ok(m + result),
            Err(f) => Err(USimpleError::new(1, format("{}", err))),,
        }
    }?;

thread::sleep(sleep_dur);

Reply:
Ah interesting, so the `?` operator can be used on a `Result` to make it return earlier. Nice! 

So I ended up with 

```
fn sleep(args: Vec<&str>) -> UResult<()> {
    let sleep_dur =
        args.iter().try_fold(
            Duration::new(0, 0),
            |result, arg| match uucore::parse_time::from_str(&arg[..]) {
                Ok(m) => Ok(m + result),
                Err(f) => Err(USimpleError::new(1, format!("{}", f))),
            },
        )?;

    Ok(thread::sleep(sleep_dur))
}
```

However, I'm unsure what the convention is here on the return. Is it okay to wrap the `thread::sleep(sleep_dur)` with an `Ok` or does it have to be separate items .ie

```
thread::sleep(sleep_dur);
Ok(())
```